src/api/app/models/package.rb

```diff
@@ -205,14 +205,21 @@ def self.get_by_project_and_name(project, package, opts = {})
       pkg = prj.packages.find_by_name(package) if pkg.nil?
     end
 
-    # FIXME: Why is this returning nil (the package is not found) if _ANY_ of the
-    # linking projects is remote? What if one of the linking projects is local
-    # and the other one remote?
     if pkg.nil? && opts[:follow_project_links]
-      # in case we link to a remote project we need to assume that the
-      # backend may be able to find it even when we don't have the package local
+      # project links can point to remote projects. In that case `p` is a String.
+      # Let's see if the backend knows about this package.
       prj.expand_all_projects.each do |p|
-        return nil unless p.is_a?(Project)
+        next unless p.is_a?(String)
+        next unless exists_on_backend?(package, prj)
+
+        # alright the backend knows this package.
+        # let's setup an in memory, read only object...
+        pkg = prj.packages.new(name: package)
```

Comment:
did you benchmark this? It can slow down the UI a lot when you need to parse plenty of packages from remote sites. Also may break the local UI if remote has temporary struggle...

I would not do this without a local cache from my gut feelings...

Reply:
> did you benchmark this? 

This only instantiates a single package. So we are talking one new `GET` request per package

> It can slow down the UI a lot when you need to parse plenty of packages from remote sites. 

I don't parse anything, I just ask the backend if it exists. But yeah I'm still checking what consequences this will have.